iac-extensions/arm/src/test/resources/checks/LogRetentionCheck/test.json

```diff
@@ -0,0 +1,170 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "name": "Raise Issue as retentionDays are too low",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true,
+          "retentionDays": 7
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as retentionDays are missing",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as isEnabled is missing,
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "retentionDays": 15
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as retentionDays are too low and isEnabled is false",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": false,
```

Comment:
I used this message `"Omitting \"isEnabled\" results in a short log retention duration. Make sure it is safe here."` for this issue. But maybe we should define a third message like:
`"Disabling \"insights\" results in a short log retention duration. Make sure it is sage here."`, when `isEnabled` is `false`

Reply:
I like the Idea, I will update RSPEC